File: test_files/feature4-developer-friendly-feedback.ts
/*
Test: Feature 4 - Developer-Friendly Feedback (TypeScript)
Contains issues with inline EXPECTED FIX comments to validate that the guardrails provide actionable remediation.
*/

export function getUserByNameUnsafe(name: string) {
  // ISSUE: SQL injection via string formatting
  // EXPECTED FIX: Use parameterized queries (e.g., db.query('SELECT * FROM users WHERE name = $1', [name]))
  const query = `SELECT * FROM users WHERE name = '${name}'`;
  return query;
}

export const API_TOKEN = 'token-1234-SECRET';
// EXPECTED FIX: use process.env['API_TOKEN'] or a secret manager

File: test_files/feature4-developer-friendly-feedback.py
"""
Test: Feature 4 - Developer-Friendly Feedback
Contains issues with inline notes describing the expected corrective action to validate that the guardrails provide actionable, fixable suggestions.
"""

# Insecure DB query - expected fix is shown in comments for the tool to reference

def get_user_by_name_unsafe(name):
    # ISSUE: SQL injection via string formatting
    # EXPECTED FIX: Use parameterized queries (e.g., db.execute("SELECT * FROM users WHERE name = %s", (name,)))
    query = "SELECT * FROM users WHERE name = '%s'" % name
    return query

# Hardcoded secret - expected fix: use env var and secret manager
API_TOKEN = "token-1234-SECRET"
# EXPECTED FIX: os.environ['API_TOKEN'] or SecretManager.get("...")

if __name__ == '__main__':
    print(get_user_by_name_unsafe("admin' OR '1'='1"))
